near-sdk/src/store/key.rs

```diff
@@ -0,0 +1,78 @@
+use borsh::BorshSerialize;
+
+use crate::env;
+
+mod private {
+    /// Seal `ToKey` implementations to limit usage to the builtin implementations
+    pub trait Sealed {}
+
+    impl Sealed for super::Sha256 {}
+    impl Sealed for super::Keccak256 {}
+    impl Sealed for super::Identity {}
+}
+
+/// Trait used to generate keys to store data based on a serializable structure.
+pub trait ToKey: self::private::Sealed {
+    /// Output type for the generated lookup key.
+    type KeyType;
```

Comment:
It's a more restrictive bound so it's generally better to have it on the impls or where it is actually needed, but since this is intended to just be used internally and it simplifies the code a lot, I've made the change